fetch.bs

```diff
@@ -8713,82 +8713,142 @@ relevant to you. The rest of the parameters are used less frequently, often for
+<a for=/>response</a>, and in particular at what stage you would like to receive a callback:
 
 <dl>
- <dt><a for=fetch><i>processRequestBodyChunkLength</i></a>
- <dd><p>Takes an algorithm that will be passed the number of bytes that have been transmitted from
- the <a for=/>request</a>'s <a for=request>body</a>. The algorithm will be invoked for each
- transmitted chunk. Most standards will not need this.
-
- <dt><a for=fetch><i>processRequestEndOfBody</i></a>
- <dd><p>Takes an algorithm that will be passed nothing. Indicates <a for=/>request</a>'s
- <a for=request>body</a> has been transmitted. Most standards will not need this.
-
- <dt><a for=fetch><i>processEarlyHintsResponse</i></a>
- <dd><p>Takes an algorithm that will be passed a <a for=/>response</a> (whose
- <a for=response>status</a> is 103). Can only be used for navigations as defined by
- <cite>HTML</cite>. [[HTML]]
-
- <dt><a for=fetch><i>processResponse</i></a>
+ <dt>Upon completion
  <dd>
-  <p>Takes an algorithm that will be passed a <a for=/>response</a>. Indicates
-  <a for=/>response</a>'s <a for=response>header list</a> has been received and initialized. This
-  is primarily useful for standards that want to operate on <a for=/>response</a>'s
-  <a for=response>body</a>'s <a for=body>stream</a> directly.
-
-  <p>If the <a for=/>request</a>'s <a for=request>mode</a> is "<code>navigate</code>" and its
-  <a for=request>redirect mode</a> is "<code>manual</code>", then callers need to follow a very
-  specific flow with this algorithm to get the intended behavior. They should compute the
-  appropriate <a for=response>location URL</a>, and if it is non-null or failure, then they should
-  call <a for="fetch controller">process the next manual redirect</a>. This will result in
-  <a for=fetch><i>processResponse</i></a> being called again, with the next <a for=/>response</a>
-  in the redirect chain.
-
- <dt><a for=fetch><i>processResponseEndOfBody</i></a>
- <dd><p>Takes an algorithm that will be passed a <a for=/>response</a>. Indicates the network is
- done transmitting the response. This does not read <a for=/>response</a>'s
- <a for=response>body</a>.
+  <p>This is the most common way in which clients handle a <a for=/>response</a>, for example
+  <a lt="fetch a classic script">scripts</a> and <a lt="fetch a style resource">style resources</a>.
+  The <a for=/>response</a>'s <a for=response>body</a> is obtained in its entirety into a
+  <a>byte sequence</a>, and then processed by the client.
+
+  <p>To process a <a for=/>response</a> upon completion, pass an algorithm as the
+  <a for=fetch><i>processResponseConsumeBody</i></a> argument of <a for=/>fetch</a>. The given
+  algorithm is passed a <a for=/>response</a> and null, failure, or a <a>byte sequence</a>.
+  The second argument would be null if there was a <a>network error</a>, failure if there was an
+  error while downloading, or a <a>byte sequence</a> representing the successfully obtained
+  <a for=response>body</a>.
```

Comment:
This seems incorrect. It's null when there is no body, failure when there's a network error, and otherwise the body as a byte sequence.

Reply:
I believe it's correct.

See [here](https://fetch.spec.whatwg.org/#ref-for-fetch-params-process-response-consume-body%E2%91%A1):

Let processBodyError be this step: run fetchParams’s [process response consume body](https://fetch.spec.whatwg.org/#fetch-params-process-response-consume-body) given response and failure.

So `failure` is a stream error and `null` is a network error.

Reply:
I see what you mean now, but that's not entirely correct. A network error has a null body and therefore it's null, but other responses can have null bodies as well.

In fact, https://fetch.spec.whatwg.org/#ref-for-process-response-end-of-body explains this quite well and this next text doesn't do it justice.

Reply:
Right, I didn't account for null-body status is this. Corrected

Reply:
Reading up on this again it seems like the request method could also be `CONNECT` or `HEAD` although that's fairly unlikely. Perhaps we should just state that response's body is null for this case and perhaps give some examples in parenthesis.

Reply:
OK I made the null cases more of a "for example" thing